addons/webhooks/clusterbootstrap_webhook.go

```diff
@@ -82,7 +83,58 @@ var _ webhook.CustomValidator = &ClusterBootstrap{}
 
 // Default implements webhook.Defaulter so a webhook will be registered for the type
 func (wh *ClusterBootstrap) Default(ctx context.Context, obj runtime.Object) error {
-	// TODO: Placeholder for future work https://github.com/vmware-tanzu/tanzu-framework/issues/1916
+	clusterBootstrap, ok := obj.(*runv1alpha3.ClusterBootstrap)
+	if !ok {
+		return apierrors.NewBadRequest(fmt.Sprintf("expected a ClusterBootstrap but got a %T", obj))
+	}
+
+	var tkrName string
+	var annotationExist bool
+	if clusterBootstrap.Annotations != nil {
+		tkrName, annotationExist = clusterBootstrap.Annotations[constants.AddCBMissingFieldsAnnotationKey]
+	}
+	if !annotationExist {
+		// It is a no-op if the annotation does not exist or the annotation value is empty
+		return nil
+	}
+
+	clusterbootstraplog.Info("attempt to add defaults", "name", clusterBootstrap.Name)
+	if tkrName == "" {
+		err := fmt.Errorf("invalid value for annotation: %s. The value needs to be the name of TanzuKubernetesRelease",
+			constants.AddCBMissingFieldsAnnotationKey)
+		clusterbootstraplog.Error(err, fmt.Sprintf("unable to add defaults to the missing fields of ClusterBootstrap %s/%s",
+			clusterBootstrap.Namespace, clusterBootstrap.Name))
+		return err
+	}
+
+	// Get the helper ready for the defaulting logic
+	helper := clusterbootstrapclone.Helper{Logger: clusterbootstraplog}
+
+	// Get ClusterBootstrapTemplate and attempt to add defaults to the missing fields of ClusterBootstrap
+	clusterBootstrapTemplateName := tkrName // TanzuKubernetesRelease and ClusterBootstrapTemplate share the same name
+	clusterBootstrapTemplate := &runv1alpha3.ClusterBootstrapTemplate{}
+	if err := wh.Client.Get(ctx, client.ObjectKey{Namespace: wh.SystemNamespace, Name: clusterBootstrapTemplateName}, clusterBootstrapTemplate); err != nil {
+		clusterbootstraplog.Error(err, fmt.Sprintf("unable to add defaults to the missing fields of ClusterBootstrap %s/%s",
+			clusterBootstrap.Namespace, clusterBootstrap.Name))
+		return err
+	}
+	if err := helper.AddMissingSpecFieldsFromTemplate(clusterBootstrapTemplate, clusterBootstrap); err != nil {
+		clusterbootstraplog.Error(err, fmt.Sprintf("unable to add defaults to the missing fields of ClusterBootstrap %s/%s",
+			clusterBootstrap.Namespace, clusterBootstrap.Name))
+		return err
+	}
+
+	// Get TanzuKubernetesRelease and attempt to complete the partial filled ClusterBootstrapPackage RefName
+	tkr := &runv1alpha3.TanzuKubernetesRelease{}
+	if err := wh.Client.Get(ctx, client.ObjectKey{Name: tkrName}, tkr); err != nil {
+		clusterbootstraplog.Error(err, fmt.Sprintf("unable to get the TanzuKubernetesRelease %s", tkrName))
+		return err
+	}
+	if err := helper.CompleteCBPackageRefNamesFromTKR(tkr, clusterBootstrap); err != nil {
```

Comment:
Should we derive package names first and then do defaulting? It will help with scenarios where we don't want to end up with the same package twice in additionalPackages.

Reply:
Done, changed the order of CompleteCBPackageRefNamesFromTKR() and AddMissingSpecFieldsFromTemplate()